modules/tinymce/src/core/main/ts/delete/TableDelete.ts

```diff
@@ -99,11 +112,30 @@ const emptyMultiTableCells = (
   endTableCells: SugarElement<HTMLTableCellElement>[],
   betweenRng: Range
 ): boolean => {
-  cleanCells(startTableCells.concat(endTableCells));
+  const rng = editor.selection.getRng();
+
+  const startCell = startTableCells[0];
+  const endCell = endTableCells[endTableCells.length - 1];
+
+  deleteContentInsideCell(startCell, rng, true);
+  deleteContentInsideCell(endCell, rng, false);
+
+  // The cursor is always collapsed back into the start cell, so we never need to clean it
+  const startTableCellsToClean = startTableCells.slice(1);
+
+  let endTableCellsToClean = endTableCells;
+  const lastCell = endTableCells[endTableCells.length - 1];
+  // Only clean empty cells, and the last cell has the potential to still have content
+  if (!editor.dom.isEmpty(lastCell.dom)) {
+    endTableCellsToClean = endTableCells.slice(0, endTableCells.length - 1);
+  }
```

Comment:
```suggestion
  const lastCell = endTableCells[endTableCells.length - 1];
  // Only clean empty cells, and the last cell has the potential to still have content
  const endTableCellsToClean = editor.dom.isEmpty(lastCell.dom) ? endTableCells : endTableCells.slice(0, -1);
```
This removes the need for the mutation.

Reply:
Whoops